x-pack/plugins/fleet/public/search_provider.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import type { CoreSetup, CoreStart } from 'src/core/public';
+
+import type { Observable } from 'rxjs';
+import { from, of, combineLatest } from 'rxjs';
+import { map, shareReplay, takeUntil } from 'rxjs/operators';
+
+import type {
+  GlobalSearchResultProvider,
+  GlobalSearchProviderResult,
+} from '../../global_search/public';
+
+import { INTEGRATIONS_PLUGIN_ID } from '../common';
+
+import { sendGetPackages } from './hooks';
+import type { GetPackagesResponse } from './types';
+import { pagePathGetters } from './constants';
+
+const packageType = 'package';
+
+export const createPackageSearchProvider = (core: CoreSetup): GlobalSearchResultProvider => {
+  const coreStart$ = from(core.getStartServices()).pipe(
```

Comment:
I am very new to RxJS - is the `$` on variables around observables a convention we've established?

Reply:
yeah, this is the convention to use when creating a variable that contains an Observable :)